### PR TITLE
Updated Vault docs, uplifted Vault image version

### PR DESCRIPTION
**Pull request by RamazanYapparov**
- changed code example in the docs to codeinclude
- removed `VaultClientTest` class and moved its test to `VaultContainerTest`
- uplifted Vault image version
- added `getVaultAddress` method for usage convenience

This PR is also related to #1158 issue

### CODE REVIEW

**Comment by vcvitaly** [comment]
I'm not a vault user but looks good at a first glance.

File: modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java [hunk 1]
```diff
@@ -29,41 +35,47 @@
...
-        assertThat(output, containsString("password4"));
+        assertThat(output).contains("password1");
+        assertThat(output).contains("password2");
+        assertThat(output).contains("password3");
+        assertThat(output).contains("password4");
     }
 
     @Test
-    public void readFirstSecretPathOverHttpApi() throws InterruptedException {
+    // readWithHttpApi {
```
Comment (rnorth) [inline modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java:62]: This comment isn't really necessary - we could have the `codeinclude` tag in the docs just say: `inside_block:readFirstSecretPathOverHttpApi`.

File: modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java [hunk 2]
```diff
@@ -29,41 +35,47 @@
...
+    public void readFirstSecretPathOverHttpApi() {
         given().
```
Comment (rnorth) [inline modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java:64]: I know the `given/when/then` style test was here originally, but would mind refactoring these  test methods to use plain style (e.g. `readFirstSecretPathOverClientLibrary`)?

It feels really strange to see `given/when/then` appearing in the documentation: 
image

**Review by rnorth — CHANGES_REQUESTED** [review]
Thanks for the contribution - a couple of minor changes but I'm grateful that you're helping to improve the docs.

File: modules/vault/src/main/java/org/testcontainers/vault/VaultContainer.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 public class VaultContainer<SELF extends VaultContainer<SELF>> extends GenericContainer<SELF> {
 
     private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("vault");
-    private static final String DEFAULT_TAG = "1.1.3";
+    private static final String DEFAULT_TAG = "1.6.1";
```
Comment (eddumelendez) [inline modules/vault/src/main/java/org/testcontainers/vault/VaultContainer.java:30]: it is not needed

**Review by eddumelendez — CHANGES_REQUESTED** [review]
I like the docs improvements and the new `getVaultAddress`. @RamazanYapparov is there a chance to rebase the branch, resolve the conflicts (for example, we already use assertj) and make the changes requested?

**Pull request merged**